centreon-dsm/www/modules/centreon-dsm/core/configuration/services/DB-Func.php

```diff
@@ -48,33 +48,43 @@ function getListServiceForPool($poolId)
 {
     global $pearDB;
 
+    if (!is_int($poolId)) {
+        return [];
+    }
+
     /*
     * Get pool informations
     */
-    $res = $pearDB->query('SELECT pool_host_id, pool_prefix FROM mod_dsm_pool WHERE pool_id = ' . $poolId);
-    $row = $res->fetch();
-    $res->closeCursor();
-
+    $stmt = $pearDB->prepareQuery('SELECT pool_host_id, pool_prefix FROM mod_dsm_pool WHERE pool_id = :pool_id');
+    $pearDB->executePreparedQuery($stmt, [':pool_id' => [$poolId, PDO::PARAM_INT]], true);
+    $row = $pearDB->fetch($stmt);
```

Comment:
Don't forget that the new methods of CentreonDb can throw a CentreonDbException, how do you manage the exceptions ? :)

Reply:
Since there are many uses of CentreonDb methods in the same functions, and each of these methods already handles exceptions internally, adding additional try-catch blocks around each call might be unnecessary. Instead, if I want to return a custom response or add specific logging, I should add a custom exception handler only where it's needed. For this file, it doesn't seem necessary to add extra error handling unless there's a specific requirement to handle an exception differently than CentreonDb already does. WDYT.